src/core/tests/model_util_tests.cpp

```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gmock/gmock.h>
+
+#include "openvino/core/model.hpp"
+#include "openvino/core/model_util.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/multiply.hpp"
+
+namespace ov::test {
+
+using op::v0::Parameter, op::v1::Add, op::v1::Multiply, op::v0::Result;
+
+class ModelUtilTest : public testing::Test {
+protected:
+    static auto make_model_without_tensor_names() {
+        auto input_1 = std::make_shared<Parameter>(element::f32, Shape{1, 3});
+        auto input_2 = std::make_shared<Parameter>(element::f32, Shape{3, 3});
+        auto input_3 = std::make_shared<Parameter>(element::f32, Shape{3, 1});
+
+        auto add = std::make_shared<Add>(input_1, input_2);
+        auto mul = std::make_shared<Multiply>(add, input_3);
+        auto add_final = std::make_shared<Add>(mul, add);
+
+        auto add_result = std::make_shared<Result>(add);
+        auto mul_result = std::make_shared<Result>(mul);
+        auto final_result = std::make_shared<Result>(add_final);
+
+        auto model = std::make_shared<Model>(ResultVector{add_result, mul_result, final_result},
+                                             ParameterVector{input_1, input_2, input_3},
+                                             "ModelWithoutTensorNames");
+        return model;
+    }
+
+    static auto make_model_with_named_nodes() {
+        auto model = make_model_without_tensor_names();
+
+        const auto& inputs = model->inputs();
+        for (size_t i = 0; i < inputs.size(); ++i) {
+            inputs[i].get_node()->set_friendly_name("input_" + std::to_string(i + 1));
+        }
+
+        auto results = model->get_results();
+        results[0]->set_friendly_name("add_result");
+        results[1]->set_friendly_name("mul_result");
+        results[2]->set_friendly_name("final_result");
+        return model;
+    }
+};
+
+TEST_F(ModelUtilTest, manual_set_all_input_tensors_names) {
+    const auto inputs_names = TensorNamesMap{{0, {"input_1"}}, {1, {"input_2"}}, {2, {"input_3"}}};
+
+    auto model = make_model_without_tensor_names();
+    util::set_input_tensors_names(*model, inputs_names);
+
+    ASSERT_EQ(model->inputs().size(), inputs_names.size());
+    for (const auto& [port, names] : inputs_names) {
+        EXPECT_EQ(model->input(port).get_names(), names) << "Names not match for input port " << port;
+    }
+}
+
+TEST_F(ModelUtilTest, manual_set_some_input_tensors_names) {
+    const auto inputs_names = TensorNamesMap{{2, {"input_2", "mul_input"}}, {0, {"add_input"}}};
+    auto expected_names = inputs_names;
+    expected_names.emplace(1, TensorNames{});
+
+    auto model = make_model_without_tensor_names();
+    util::set_input_tensors_names(*model, inputs_names);
+
+    ASSERT_EQ(model->inputs().size(), expected_names.size());
+    for (const auto& [port, names] : expected_names) {
+        EXPECT_EQ(model->input(port).get_names(), names) << "Names not match for input port " << port;
+    }
```

Comment:
It looks like repetitive piece of code (same for outputs).

Reply:
?

Reply:
Such check (on different parameters) is also in other test cases in the file.

Reply:
Yes to have fail in exact place in test.

Reply:
Fair enough.

Reply:
The test helper added to compare tensors, and error message has been updated also.